### ISSUE TITLE
Project Ideas

### ISSUE THREAD

**Issue by CKolkey** [FabijanZulj/blame.nvim#2]
Just some initial impressions:
- In the blame window, I would expect `<esc>` and `q` to close it.
- You can use an autocommand on the `BufUnload` to ensure `https://github.com/FabijanZulj/blame.nvim/blob/main/lua/blame/window_blame.lua#L40` gets called.
- Navigation through the blame-stack a-la github by hitting `<cr>` on the specific line

Great work - looking forward to seeing this progress :)

**Comment by FabijanZulj** [comment]
Thank you for the comment.

- I agree, I will add so it closes on esc and q
- I am on it also :)
- I think I understand. When you CR on a certain commit in the blame window it would show the file and blame prior to that commit?
I will have to think about how to achieve this but it sounds very cool and Ill work towards it!

I really appreciate your input :)

**Comment by skoch13** [comment]
Thank you for the plugin. It might be also helpful to provide integration with https://github.com/sindrets/diffview.nvim as it provides diffs for regions

**Comment by CKolkey** [comment]
> - I think I understand. When you CR on a certain commit in the blame window it would show the file and blame prior to that commit?

Yeah, think of like like a link to the file at that commit. So not necessarily the prior commit, but the one for that blame. I know diffview can do it, though I think thats by using a tab-page and temp file instead of updating the repo itself.

**Comment by FabijanZulj** [comment]
I will start working on this "blame stack" feature.
What my idea is:

Pressing TAB on a blame line - opens the `git show {hash}:{file_currently_open}` in a 
- floating window,
- a new window 
- or replaces the content of the buffer you are blaming with the file at that commit 
Depending on an option (or a different keybind)

If the content of the blame buffer is replaced (last option), the blame window is updated with "git blame" at that state.  

Pressing 'backspace' moves back through the stack and updates the buffer and blame window until you are at the initial state of the file.

I am not sure if that was the initial idea. So if anyone has any comments that would be great :)

**Comment by thecontinium** [comment]
Thanks for the plugin - I would like to see the last "replace" option you mention above; being able to go up/down the blame commit stack in the same window with an option to exit blame and be at that commit.

**Comment by FabijanZulj** [comment]
I have this implemented basically.
However, there is one issue. It is with file renames.
Since `git show` doesn't follow file renames, it cannot be used alone in every case.
I am not sure if there is an easy way to achieve this instead of parsing `git log --follow` and then with that path name executing
`git show`

**Comment by CKolkey** [comment]
Congrats with the refresh :) The blame stack is pretty slick. Should it be possible to push to the stack without needing to be in the blame window?

**Comment by FabijanZulj** [comment]
Thanks! :) 
Yes I just added it, thought about it but must have forgotten along the way

**Comment by sathishmanohar** [comment]
A command/keyboard shortcut to copy commit hash or short hash to the clipboard from the blame split.

**Comment by elmehalawi** [comment]
It would be really great to show the color based on how long ago the commit was, so the darker the color the more recent the commit is. Github's blame view does this and I find it to be very useful since 99% of the time when I'm in this view it's to find the cause of something that happened recently.

**Comment by Varal7** [comment]
This blame stack is pretty sweet! It's the only feature I use in fugitive.vim and couldn't get it to work in other git plugins.

The way it works in in Fugitive is I `:Git blame` then <CR> on the commit, then <CR> on a line from the parent commit. This populates a buffer where the file looks like that parent commit. And I can `:Git blame` on that buffer again, etc. 
I can easily navgiate back and forth with `C-O` and `C-I`.

https://github.com/user-attachments/assets/69eb0432-aa6b-45e1-b61f-4a29e894a610

With blame.nvim, I can use TAB and backspace to move back and forth but if I decide to open the commit, I lose the whole context of my stack. 

https://github.com/user-attachments/assets/1fd96f3c-11cb-4f0e-9f9a-6e906b16dd94

Any thoughts on how to fix?
